src/Compilers/CSharp/Portable/CSharpResources.resx

```diff
@@ -7839,4 +7839,16 @@ To remove the warning, you can use /reference instead (set the Embed Interop Typ
   <data name="IDS_ImplicitIndexerInitializer" xml:space="preserve">
     <value>implicit indexer initializer</value>
   </data>
+  <data name="WRN_ConvertingLock" xml:space="preserve">
+    <value>A value of type 'System.Threading.Lock' converted to a different type will use likely unintended monitor-based locking in 'lock' statement.</value>
+  </data>
+  <data name="WRN_ConvertingLock_Title" xml:space="preserve">
+    <value>A value of type 'System.Threading.Lock' converted to a different type will use likely unintended monitor-based locking in 'lock' statement.</value>
+  </data>
+  <data name="ERR_BadSpecialByRefLock" xml:space="preserve">
+    <value>A lock statement on a value of type 'System.Threading.Lock' cannot be used in async methods or async lambda expressions.</value>
+  </data>
+  <data name="IDS_NativeLock" xml:space="preserve">
+    <value>native lock</value>
```

Comment:
>native lock [](http://example.com/codeflow?start=11&length=11)

I am not sure what "native" is supposed to convey to a user. Even I am not sure why we call the feature this way. Consider calling  it a "'Lock' object" instead.